### PR TITLE
Fixed SSL support with treq

### LINKED ISSUE #3441: NotImplementedError: SSL support unavailable
From reporter (Tribler 7.0.0)
> I added an RSS feed and now Tribler crashes on boot every single time

```python
Traceback (most recent call last):
File "TriblerGUI\event_request_manager.py", line 99, in on_read_data
RuntimeError: Unhandled Error
Traceback (most recent call last):
File "lib\multiprocessing\process.py", line 114, in run

File "TriblerGUI\core_manager.py", line 65, in start_tribler_core

File "lib\site-packages\twisted\internet\base.py", line 1243, in run

File "lib\site-packages\twisted\internet\base.py", line 1252, in mainLoop

--- <exception caught here> ---
File "lib\site-packages\twisted\internet\base.py", line 878, in runUntilCurrent

File "Tribler\Core\Modules\channel\channel_rss.py", line 114, in _task_scrape

File "Tribler\Core\Modules\channel\channel_rss.py", line 111, in parse_feed

File "Tribler\Core\Modules\channel\channel_rss.py", line 267, in parse

File "Tribler\Core\Utilities\utilities.py", line 215, in http_get

File "lib\site-packages\twisted\web\client.py", line 1633, in request

File "lib\site-packages\twisted\web\client.py", line 1617, in _getEndpoint

File "lib\site-packages\twisted\web\client.py", line 1494, in endpointForURI

File "lib\site-packages\twisted\web\client.py", line 857, in raiseNotImplemented

exceptions.NotImplementedError: SSL support unavailable
```

### PR DESCRIPTION
**Pull request by xoriole**
Fixes #3441 : NotImplementedError: SSL support unavailable

### CODE REVIEW

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

File: Tribler/Test/Core/test_utilities.py [hunk 1]
```diff
@@ -236,4 +237,34 @@ def test_add_url_param_clean(self):
...
+        uri = "https://example.com"
 
+        def cbResponse(response):
+            self.assertIsNotNone(response)
+
+        def cbErrorResponse(_):
+            self.fail("Error was not expected.")
+
+        http_deferred = http_get(uri)
+        http_deferred.addCallback(cbResponse)
```
Comment (qstokkink) [inline Tribler/Test/Core/test_utilities.py]: @devos50 was this O.K. or does this need to be `addCallbacks()`?

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Pull request merged**